### PR TITLE
Add benchmark for feeding at SPI layer.

### PR DESCRIPTION
**Pull request by toregge**
@baldersheim : please review

### CODE REVIEW

**Comment by baldersheim** [comment]
@geirst and @vekterli Please also take a look

**Review by baldersheim — APPROVED** [review]
This looks good as an initial start.
I would guess that the following improvements should be considered.
1 - Documents should made pre generated and serialised and then being created from serialised form to mimic how it happens in a live system. Due to lazyness deserialisation does not happen on the way to the TLS, but when forked off to the writer threads.
2 - Parameterise some of the test values. Number of unique documents to feed, number of refeeds, number of spi threads, etc.

**Review by geirst — COMMENTED** [review]
👍

File: searchcore/src/apps/vespa-spi-feed-bm/vespa_spi_feed_bm.cpp [hunk 1]
```diff
@@ -0,0 +1,464 @@
...
+    
+    PersistenceProviderFixture();
+    ~PersistenceProviderFixture();
+    void create_document_db();
+    uint32_t num_buckets() const { return (1u << _bucket_bits); }
+    Bucket make_bucket(uint32_t i) const { return Bucket(document::Bucket(_bucket_space, BucketId(_bucket_bits, i & (num_buckets() - 1))), PartitionId(0)); }
+    BucketId get_bucket_id(const Document &document) const {
+        auto &id = document.getId();
+        auto &gid = id.getGlobalId();
+        return BucketId(_bucket_bits, gid.convertToBucketId().getRawId()).stripUnused();
```
Comment (vekterli) [inline searchcore/src/apps/vespa-spi-feed-bm/vespa_spi_feed_bm.cpp:253]: Could also use `document::BucketIdFactory` to do this

**Review by vekterli — COMMENTED** [review]
:+1:

**Pull request merged**